src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -53,14 +78,20 @@ public Optional<Certificate> getCertificateFromPem(String certificatePem) throws
      *
      * @param certificatePem Certificate PEM
      * @return certificate object
+     *
      * @throws InvalidCertificateException if certificate PEM is invalid
+     * @throws CertificateException - If fails generate certificate from PEM
+     * @throws KeyStoreException - If fails to store the key store into disk
      */
-    public Certificate getOrCreateCertificate(String certificatePem) throws InvalidCertificateException {
+    public Certificate getOrCreateCertificate(String certificatePem) throws InvalidCertificateException,
+            CertificateException, KeyStoreException {
         Certificate newCert = Certificate.fromPem(certificatePem);
         Optional<CertificateV1DTO> dto = runtimeConfiguration.getCertificateV1(newCert.getCertificateId());
         if (dto.isPresent()) {
             return certificateV1DTOToCert(dto.get());
         }
+
+        this.pemStore.storePem(newCert.getCertificateId(), certificatePem);
```

Comment:
I decided not to do it with events for now. I will re evaluate if I want to refactor it. Right now it is tied to the registry so I am ok with putting it here for now and later creating a listener etc etc that stores this separately.

Reply:
To me this seems like core registry functionality. The fact that we're storing different bits of information in different places is an implementation detail. From an external perspective, this is just certificate information and it shouldn't matter where it goes. I'm also ok with it.